.github/workflows/ci-repo.yml

```diff
@@ -0,0 +1,82 @@
+name: CI (repo level)
+
+on:
+  push:
+    branches:
+      - "main"
+      - "release-*"
+
+  pull_request:
+    branches:
+      - "main"
+      - "release-*"
+
+  workflow_dispatch:
+
+  workflow_call:
+
+concurrency:
+  group: ci-repo-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check-out repository
+        uses: actions/checkout@v4
+
+      - name: Use Node.js from .nvmrc
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+
+      - name: Cache extension node_modules
+        uses: actions/cache@v3
+        with:
+          path: vscode/node_modules
+          key: ${{ runner.os }}-node-${{ hashFiles('vscode/package-lock.json') }}
+
+      - name: Verify package-lock.json
+        run: ./scripts/verify_lock.mjs
```

Comment:
Speaking from ignorance, do we need this script? Could we not just run `npm ci` and verify package-lock.json remains unchanged?

Reply:
Short answer is cachito.

This script was created to make sure the lock file will still work when built downstream.  There is a weird edge case where the lockfile can be ok for npm, but not be ok for cachito.

If no downstream build .... it isn't really super useful, but is a nice to have.

Reply:
@sjd78 @djzager I've run into problems with the script - mainly due to the fact we have kind-of-monorepo here. 
Since we have no downstream I've switched to `npm ci` for now.